src/main/scala/com/horizon/exchangeapi/BusinessRoutes.scala

```diff
@@ -864,68 +872,174 @@ trait BusinessRoutes extends JacksonSupport with AuthenticationSupport {
-              logger.debug("POST /orgs/" + orgid + "/business/policies/" + policy + "/search: looking for nodes w/o agreement for '" + searchSvcUrl)
-              if (list.nonEmpty) {
-                // Go thru the rows and build a hash of the nodes that do NOT have an agreement for our service
-                //todo: factor in num agreements??
-                val nodeHash = new MutableHashMap[String, BusinessPolicySearchHashElement] // key is node id, value noAgreementYet which is true if so far we haven't hit an agreement for our service for this node
-                for ((nodeid, nodeType, publicKey, agrSvcUrlOpt, stateOpt) <- list) {
-                  //logger.debug("nodeid: "+nodeid+", agrSvcUrlOpt: "+agrSvcUrlOpt.getOrElse("")+", searchSvcUrl: "+searchSvcUrl+", stateOpt: "+stateOpt.getOrElse(""))
-                  val nt = if (nodeType == "") NodeType.DEVICE.toString else nodeType
-                  nodeHash.get(nodeid) match {
-                    // This node is already in the hash. Only replace it if this is an agreement for the service, because the absence of an agr for this svc isn't useful info
-                    case Some(_) => if (agrSvcUrlOpt.getOrElse("") == searchSvcUrl && stateOpt.getOrElse("") != "") {
-                      /*logger.debug("setting to false");*/ nodeHash.put(nodeid, BusinessPolicySearchHashElement(nt, publicKey, noAgreementYet = false))
-                    } // this is no longer a candidate
-                    // This node is not yet in the hash. Add it with whatever value it has for agreement - this may be overridden later
-                    case None => val noAgr = if (agrSvcUrlOpt.getOrElse("") == searchSvcUrl && stateOpt.getOrElse("") != "") false else true
-                      nodeHash.put(nodeid, BusinessPolicySearchHashElement(nt, publicKey, noAgr)) // this node not in the hash yet, add it
-                  }
-                }
-                // Convert our hash to the list response of the rest api
-                //val respList = list.map( x => BusinessPolicyNodeResponse(x._1, x._2, x._3)).toList
-                val respList = new ListBuffer[BusinessPolicyNodeResponse]
-                for ((k, v) <- nodeHash) if (v.noAgreementYet) respList += BusinessPolicyNodeResponse(k, v.nodeType, v.publicKey)
-                val code = if (respList.nonEmpty) HttpCode.POST_OK else HttpCode.NOT_FOUND
-                (code, PostBusinessPolicySearchResponse(respList.toList, 0))
-              } else {
-                (HttpCode.NOT_FOUND, PostBusinessPolicySearchResponse(List[BusinessPolicyNodeResponse](), 0))
+            case Success(results) =>
+              if(results._2.nonEmpty) {
+                (HttpCode.POST_OK,
+                  PostBusinessPolicySearchResponse(
+                    results._2.map(
+                      node =>
+                        BusinessPolicyNodeResponse(
+                          node._1,
```

Comment:
Can you add some comments just anywhere there's a `._1` or `._2` to specify what the column actually is?